sigstore/_cli.py

```diff
@@ -734,6 +838,97 @@ def _verify(args: argparse.Namespace) -> None:
             sys.exit(1)
 
 
+def _verify_github(args: argparse.Namespace) -> None:
+    # Every GitHub verification begins with an identity policy,
+    # for which we know the issuer URL ahead of time.
+    # We then add more policies, as configured by the user's passed-in options.
+    inner_policies: list[policy.VerificationPolicy] = [
+        policy.Identity(
+            identity=args.cert_identity,
+            issuer="https://token.actions.githubusercontent.com",
+        )
+    ]
+
+    if args.workflow_trigger:
+        inner_policies.append(policy.GitHubWorkflowTrigger(args.workflow_trigger))
+    if args.workflow_sha:
+        inner_policies.append(policy.GitHubWorkflowSHA(args.workflow_sha))
+    if args.workflow_name:
+        inner_policies.append(policy.GitHubWorkflowName(args.workflow_name))
+    if args.workflow_repository:
+        inner_policies.append(policy.GitHubWorkflowRepository(args.workflow_repository))
+    if args.workflow_ref:
+        inner_policies.append(policy.GitHubWorkflowRef(args.workflow_ref))
+
+    policy_ = policy.AllOf(inner_policies)
+
+    verifier, files_with_materials = _collect_verification_state(args)
+    for (file, materials) in files_with_materials:
+        result = verifier.verify(materials=materials, policy=policy_)
+
+        if result:
```

Comment:
NB: From here to the end is pretty much just a dupe of the body of `_verify_identity`, so it isn't very DRY of us. It turns out that the two are nearly identical except for how the various verification flags are handled, so I need to think some more about whether it really makes sense for them to have separate functions like this.

Reply:
Yeah, it feels like a lot of this could be wrapped up into a function. It seems like it's mostly the beginning before the `(file, materials)` loop that differs so this should separate pretty cleanly.